### ISSUE TITLE
IE 11 errors when optimization is set to true

### ISSUE THREAD

**Issue by karptonite** [angular/angular#32379]
# 🐞 bug report

### Affected Package

 The issue is caused by package @angular/cli

### Is this a regression?

 Yes, the previous version in which this bug was not present was: 7

### Description
When our app is build with the CLI setting `"optimization": true`, it breaks on IE 11, despite having the correct setting for IE 11 in `browserslist` and the correct polyfills.

The error message is variously in the form: `Error: Uncaught (in promise): TypeError: Object.create: argument is not an Object and is not null.`, and `ERROR Error: Uncaught (in promise): TypeError: Unable to get property 'enumerable' of undefined or null reference`. I can't tell exactly where the problem originates, but the immediate issue appeas to be somerwhere in polyfills. If i turn off `optimization`, it works fine, but of course, then optimization is off for ALL users.

## 🔬 Minimal Reproduction

 
I wish I could create a minimal repro of this. I haven't been able to, but if I get feedback that this is a serious issue (rather than, for example "`optimization` is known to be broken on IE"), I will to give it another try. You can see the problem in action now by loading [this page](https://boardgamegeek.com/image/100/stephensons-rocket) in IE. You can also see the same problem in older versions of Firefox (52) and Chrome(49);  those aren't even nominally supported by Angular, but I mention them because the dev tools may be better. I do know that the problem doesn't occur on every page of our app--some of the more simple pages on our site, including  [this one](https://boardgamegeek.com/join) seem to work OK, so the issue will not pop up on minimal site.

## 🔥 Exception or Error
One error I've seen:
<pre><code>
Error: Uncaught (in promise): TypeError: Object.create: argument is not an Object and is not null.
</code></pre>
And another:
<pre><code>
ERROR Error: Uncaught (in promise): TypeError: Unable to get property 'enumerable' of undefined or null reference
</code></pre>

## 🌍  Your Environment
Not sure the environment is relevant, since this is an issue in production, but I work and build the code on MacOS Mojave, but for production, the code is build on CentOS.

**Angular Version:**
<pre><code>
@angular-devkit/architect         0.803.0
@angular-devkit/build-angular     0.803.0
@angular-devkit/build-optimizer   0.803.0
@angular-devkit/build-webpack     0.803.0
@angular-devkit/core              8.2.2
@angular-devkit/schematics        8.2.2
@angular/cdk                      8.1.3
@angular/cli                      8.3.0
@ngtools/webpack                  8.3.0
@schematics/angular               8.3.0
@schematics/update                0.803.0
rxjs                              6.5.2
typescript                        3.5.3
webpack                           4.39.2
</code></pre>

**Anything else relevant?**

Definitely broken only in IE 11 (of supported browsers), as well as Firefox 52 and Chrome 49 (of unsupported browsers). Thinking about it more, it may be a problem for all browsers which load ES5.

**Comment by clydin** [comment]
Duplicate of https://github.com/angular/angular-cli/issues/15436 and will be fixed via https://github.com/angular/angular-cli/pull/15477

As a temporary workaround until the next patch release, you can set an environment variable `NG_BUILD_DIFFERENTIAL_FULL` to `1` or `true` prior to executing the build command.

**Comment by karptonite** [comment]
Thank you thank you thank you, @clydin! You saved me from many additional hours of trying to create my own repro, and, hopefully, prevented a full-scale revolt by the small minority of our users who were affected by this bug.

**Comment by clydin** [comment]
Angular CLI 8.3.2 has been released.  If you are still encountering the issue, please let us know.

**Comment by nnit-TroelsJensen** [comment]
Fantastic. I tried upgrading two days ago and got stuck with this IE 11 + prod build problem. I thought it was a polyfill thing and tried almost everything.
Then i saw version 8.3.1 earlier today and thought maybe its fixed, or i did something wrong the first time... Let me try again.
Same problem.... i almost gave up when i stumbled on this bug report and... weeee 

> Angular CLI 8.3.2 has been released
23 minutes ago 

😄 Thx
Hopefully this also fixes the memory issue for larger projects 🤞

**Comment by karptonite** [comment]
@clydin I pushed this out today without the `NG_BUILD_DIFFERENTIAL_FULL` workaround. While it seemed to help (I no longer had the problem on IE 11), there were still issues on Firefox 52, which I'm aware is unsupported by Angular, but which is still used by users who are stuck on Windows Vista. I didn't test any of the other older browsers, and just put back in the `NG_BUILD_DIFFERENTIAL_FULL` for now. The error was in the form:
```
 Error: Uncaught (in promise): TypeError: e is not an object or null
.pDpN/</Object.create@https://cf.geekdo-static.com/geekui2/polyfills-es5.f0025e2cd5d100539752.js:1:120382
e@https://cf.geekdo-static.com/geekui2/4-es5.f58a3466e425995f0e02.js:1:263
t/g<@https://cf.geekdo-static.com/geekui2/3-es5.b5fc2af1f76a4da540de.js:1:35628
```
Etc, etc. 

I have a system where I can replicate this issue locally, but no minimal repro. If this is something that the Angular team would like to address, I'm willing to help. I realize that this is for an officially unsupported browser, but the fact that the page works perfectly with the `NG_BUILD_DIFFERENTIAL_FULL` workaround suggests that there may be real bug in the conversion from ES2015 to ES5, I assume, and it may affect some supported browser I have not yet tested.

**Comment by karptonite** [comment]
Hold up: That last issue might be mine. I think I had my browserslist file set to strict.

**Comment by karptonite** [comment]
Nope, still broken on Firefox 49 when the workaround is not in there.  I'd say it is possible I'm missing a polyfill, but that seems unlikely because it works with the NG_BUILD_DIFFERENTIAL_FULL=1.

**Comment by Alex-Torres** [comment]
Am running Angular CLI 8.3.1, and am having similar issues:

ERROR Error: Uncaught (in promise): TypeError: Unable to get property 'enumerable' of undefined or null reference

TypeError: Unable to get property 'enumerable' of undefined or null reference

**Comment by karptonite** [comment]
@Alex-Torres there was a bug in 8.3.1 that was fixed in 8.3.2. Try upgrading and see if that fixes your issue.

**Comment by Alex-Torres** [comment]
Sweet!  Will do thanks @karptonite

**Comment by Alex-Torres** [comment]
Still having same problem after updating to 8.3.2 🤔

**Comment by karptonite** [comment]
OK, so here is where I am: After seeing the issue in production, I rolled back to the `NG_BUILD_DIFFERENTIAL_FULL=1` for a while. But I really struggled to reproduce the issue in our dev environment. I finally tried deploying to production again without the `NG_BUILD_DIFFERENTIAL_FULL=1` hack, and am no longer seeing the issue in production either. Unless I see further issues, I'm going to chalk up the production issues to some cache that wasn't getting cleared or something. If I have any additional problems, I'll open a new issue on angular/cli. @Alex-Torres , if you are still seeing this issue, especially if you can reproduce it, I suggest you open a new issue in the angular/angular-cli project.
